src/components/decimal/decimal.component.js

```diff
@@ -227,17 +227,24 @@ const Decimal = ({
     prevControlledRef.current = isControlled;
   }, [isControlled]);
 
+  const prevValue = usePrevious(value);
+  const componentShouldUpdate = stateValue === value;
+
   useEffect(() => {
     const unformattedValue = toStandardDecimal(stateValue);
 
     if (isControlled) {
       const valueProp = getSafeValueProp(value);
       if (unformattedValue !== valueProp) {
-        setStateValue(formatValue(value));
+        setStateValue(formatValue(valueProp));
```

Comment:
```suggestion
      if (valueProp === "" && prevValue === "") {
          setStateValue(formatValue(emptyValue));
        } else {
          setStateValue(formatValue(valueProp));
        }
```

So we're not potentially setting state twice 👍